src/templater.rs

```diff
@@ -204,21 +204,24 @@ where
     T: Template<C>,
 {
     fn format(&self, context: &C, formatter: &mut dyn Formatter) -> io::Result<()> {
-        // TemplateProperty may be evaluated twice, by has_content() and format().
-        // If that's too expensive, we can instead create a buffered formatter
-        // inheriting the state, and write to it to test the emptiness. In this case,
-        // the formatter should guarantee push/pop_label() is noop without content.
```

Comment:
This comment seems unaddressed.  I think FormatRecorder could support this pretty succinctly by removing the previous LabelOp iff a LabelOp::Pop is pushed, pos hasn't changed, and the previous op is a PushLabel().

Reply:
> the formatter should guarantee push/pop_label() is noop without content.

`FormatRecorder` doesn't process labels by itself, so its `data().is_empty()` has this property, and that's all the `separate()` function needs. In later pass, the output formatter could emit redundant escape sequences, but we don't have to care.

Reply:
Still, it might be nice if FromatRecorder pre-deleted the redundant labels and saved some work downstream for empty label bodies.  I agree it's not important though.

Reply:
Yeah. Somewhat related: I'm not pretty sure which data structure is useful: `(data, [(pos, Push|Pop)])` vs `[Data(chunk)|Push|Pop]`. For the latter, eliminating empty Push/Pop is more important.

I'll revisit it when implementing text wrapping functions.